File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Go/s867221485_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	var N int
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	fmt.Sscanf(scanner.Text(), "%d", &N)
	scanner.Scan()
	s := scanner.Text()

	var ans, sha, dot int
	first := true

	for i := 0; i < len(s); i++ {
		c := s[i]
		if c == '#' {
			first = false
			if sha <= dot {
				ans += sha
				sha = 0
				dot = 0
				first = true
			}
			sha++
		} else {
			dot++
		}
	}

	if sha <= dot {
		ans += sha
		dot = 0
	}

	fmt.Println(ans + dot)
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/C/s867221485_comments_removed.c
#include <stdio.h>
#include <string.h>

int main() {
    int N;
    scanf("%d", &N);
    getchar();
    char s[100001];
    fgets(s, sizeof(s), stdin);
    s[strcspn(s, "\n")] = '\0';
    int ans = 0, sha = 0, dot = 0;
    int first = 1;
    for (int i = 0; i < N; i++) {
        if (s[i] == '#') {
            first = 0;
            if (sha <= dot) {
                ans += sha;
                sha = 0;
                dot = 0;
                first = 1;
            }
            sha++;
        } else {
            dot++;
        }
    }
    if (sha <= dot) {
        ans += sha;
        dot = 0;
    }
    printf("%d\n", ans + dot);
    return 0;
}
// End of Code
